modules/experiment/src/main/java/com/intuit/wasabi/experiment/impl/ExperimentsImpl.java

```diff
@@ -146,54 +146,40 @@ public Experiment getExperiment(Application.Name appName, Experiment.Label label
-        // delete the experiment from cassandra too.
+        String errorMessage = "Exception occurred while creating an exception in MySQL...";
         try {
-            // Append the newly created experiment to the priority list
-            priorities.appendToPriorityList(newExperimentID);
-        } catch (Exception e) {
-            // Erase the experiment from cassandra
-            cassandraRepository.deleteExperiment(newExperiment);
-            throw e;
-        }
 
-        //3. Creation in MySQL
-        // For consistency,  if any exception is raised while creating an experiment in mysql
-        // delete the experiment from cassandra too.
-        try {
+            //Step#2: Create experiment in MySQL
+            LOGGER.debug("Creating an experiment in MySQL...");
             databaseRepository.createExperiment(newExperiment);
-        } catch (Exception e) {
-            // Remove from priority list
-            priorities.removeFromPriorityList(newExperiment.getApplicationName(), newExperimentID);
-            // Erase the experiment from cassandra
-            cassandraRepository.deleteExperiment(newExperiment);
-            throw e;
-        }
 
-        //4. Creation of indices in Cassandra
-        try {
-            //only when the creation in Cassandra and MySQL succeeds should
-            // we create the entries in the index tables in cassandra
-            cassandraRepository.createIndicesForNewExperiment(newExperiment);
-        } catch (RepositoryException e) {
-            //Roll back everything (Note: I guess this case is highly unlikely to happen)
-            priorities.removeFromPriorityList(newExperiment.getApplicationName(), newExperimentID);
-            // Erase the experiment from cassandra
-            cassandraRepository.deleteExperiment(newExperiment);
-            // Erase from MySQL
-            databaseRepository.deleteExperiment(newExperiment);
-            throw e;
+            //Step#3: Create experiment in Cassandra
+            try {
+                cassandraRepository.createExperiment(newExperiment);
+            } catch (Exception exceptionFromCassandra) {
+                errorMessage = "Exception occurred while creating an experiment in Cassandra...";
+                // Erase from MySQL
+                try {
+                    databaseRepository.deleteExperiment(newExperiment);
+                } catch (Exception mysqlRollbackException) {
+                    LOGGER.error("An attempt to rollback of experiment in MySQL is failed...", mysqlRollbackException);
+                }
+                throw exceptionFromCassandra;
+            }
+
+            //Step#4: Log experiment creation event
+            eventLog.postEvent(new ExperimentCreateEvent(user, newExperiment));
+
+        } catch (Exception experimentCreateException) {
+            LOGGER.error("Error Message={}... Experiment={}, UserInfo={}", errorMessage, newExperiment, user, experimentCreateException);
```

Comment:
@abhishekjain88 you are right, I might have over complicated here trying to make code less... Removed errorMessage variable and kept direct exception logging..